aspects/aspects.go

```diff
@@ -618,6 +669,63 @@ func replaceIn(path, key, value string) string {
 	return strings.Join(parts, ".")
 }
 
+func prunePathInValue(parts []string, val interface{}) (interface{}, error) {
```

Comment:
does this merit dedicated unit tests?

Reply:
I thought TestAspectSetErrorIfValueContainsUnusedParts was able to target it fairly well, so I didn't feel the need to export it in `export_test`. I guess the disadvantage of going through the entire Set is that is not as simple as it could be